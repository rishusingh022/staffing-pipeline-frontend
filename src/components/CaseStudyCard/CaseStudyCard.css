.case-study-card-style img{
    height: 120px !important;
    object-fit: fill;
}
.case-study-card-style{
    height: 240px;
}