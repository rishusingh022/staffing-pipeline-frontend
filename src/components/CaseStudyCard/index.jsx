import React from 'react';
import Image from '../Image';
import './CaseStudyCard.css';
import { BsArrowRight } from 'react-icons/bs';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

export default function CaseStudyCard({ imageUrl, altText, identityNumber, name, author, boxLink }) {
  return (
    <div className="shadow-lg flex flex-col w-64 case-study-card-style" data-testid="image-card">
      <Image imageUrl={imageUrl} altText={altText} />
      <div className="px-3 py-1">
        <div className="text-xs mb-1">{identityNumber}</div>
        <div className="font-bold text-xl mb-1">{name}</div>
        <div className="font-medium text-xs mb-1 text-gray-500">by {author}</div>
      </div>
      <Link to={boxLink} className="link-button self-end text-xs mb-2 font-medium cursor-pointer" target="_blank">
        Read More
        <BsArrowRight className="inline-block mx-2 text-blue-800" />
      </Link>
    </div>
  );
}

CaseStudyCard.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  altText: PropTypes.string.isRequired,
  identityNumber: PropTypes.string.isRequired,
  name: PropTypes.string.isRequired,
  author: PropTypes.string.isRequired,
  boxLink: PropTypes.string.isRequired,
};
