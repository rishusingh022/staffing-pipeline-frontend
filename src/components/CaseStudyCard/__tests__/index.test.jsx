import React from 'react';
import { render } from '@testing-library/react';
import CaseStudyCard from '..';

const mockedLink = jest.fn();
jest.mock('react-router-dom', () => ({
  Link: () => mockedLink,
}));

describe('CaseStudyCard', () => {
  it('should render correctly', () => {
    const { asFragment } = render(<CaseStudyCard />);
    expect(asFragment()).toMatchSnapshot();
  });
});
