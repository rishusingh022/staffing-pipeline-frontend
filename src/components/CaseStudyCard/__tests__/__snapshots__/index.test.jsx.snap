// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CaseStudyCard should render correctly 1`] = `
<DocumentFragment>
  <div
    class="shadow-lg flex flex-col w-64 case-study-card-style"
    data-testid="image-card"
  >
    <img
      class="image-component"
    />
    <div
      class="px-3 py-1"
    >
      <div
        class="text-xs mb-1"
      />
      <div
        class="font-bold text-xl mb-1"
      />
      <div
        class="font-medium text-xs mb-1 text-gray-500"
      >
        by 
      </div>
    </div>
  </div>
</DocumentFragment>
`;
